internal/fbsemantic/semantic.fbs

```diff
@@ -446,4 +446,9 @@ table UnsignedIntegerLiteral {
   value:uint64;
 }
 
+table Compiled {
```

Comment:
Instead of adding a new root type, can you add `polytype:PolyType` as an attribute to `table Package`? This should be backwards compatible.

Reply:
I anticipate we will add more and more information to this Compiled type as we use it more. For example we may need version information. A package has meaning outside the context that is intended to be reused for compilation. I do not want to conflate those concepts.

Reply:
I would not interpret the type of the package to be something to maintain compiled Flux code on the filesystem. It seems to me to be directly associated with the package itself. I would agree that the package type is pure code that is compiled which, in my view, includes the types because it includes the type in multiple other places in the flatbuffers.

I would not agree with the assertion that this is metadata that is not required to understand the code. It's an open bug that we don't include this information for the Go code to access which causes the Go code to operate incorrectly in certain circumstances.

For the example of a hash of the source code, I would likely agree that is not something directly associated with the package. I also don't think that's a reason to have the polytype be associated with a container that holds the package instead of the package itself. I don't know if this would be something we even put into flatbuffers though.

> In other words I see the Package type as what is needed for the runtime to execute the code and I see the Compiled type as what is needed to be able to compile and persist the code.

If this is the case, I'm stronger on including the polytype in the package itself. The runtime is in desperate need of the polytype to correctly execute the code. This would give us access to that information and we would be able to make use of this inside of the compiler. This would instantly enable us to begin work on this improvement while also being backwards compatible in a way that doesn't require us to urgently work on this.